packages/react/src/components/Masthead/HeaderNavContainer.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright IBM Corp. 2016, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import calculateTotalWidth from '@carbon/ibmdotcom-utilities/es/utilities/calculateTotalWidth/calculateTotalWidth';
+import CaretLeft20 from '@carbon/icons-react/es/caret--left/20';
+import CaretRight20 from '@carbon/icons-react/es/caret--right/20';
+import PropTypes from 'prop-types';
+import settings from 'carbon-components/es/globals/js/settings';
+
+const { prefix } = settings;
+
+/**
+ * Header nav container component.
+ */
+const HeaderNavContainer = ({ children }) => {
+  const headerNavContainerRef = useRef(null);
+  const [resizeObserver, setResizeObserver] = useState(null);
+  const [showLeftCaret, setShowLeftCaret] = useState(false);
+  const [showRightCaret, setShowRightCaret] = useState(false);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [totalNavWidth, setTotalNavWidth] = useState(0);
+
+  const paginateLeft = useCallback(() => {
+    headerNavContainerRef.current.scrollLeft -= containerWidth;
+    setShowRightCaret(true);
+    if (headerNavContainerRef.current.scrollLeft <= 40) {
+      setShowLeftCaret(false);
+      headerNavContainerRef.current.scrollLeft = 0;
+    }
+  }, [containerWidth]);
+
+  const paginateRight = useCallback(() => {
+    headerNavContainerRef.current.scrollLeft += containerWidth;
+    setShowLeftCaret(true);
+    if (
+      headerNavContainerRef.current.scrollLeft + containerWidth >=
+      totalNavWidth - 80
```

Comment:
Wondering where `80` number came from?

Reply:
This 80 is coming from each caret being displayed and needing to adjust. This use case comes into play when there are 3 sections of scrolling. Please let me know if you think there is a better way of handling this.

Reply:
OK given how long this PR has been pending, probably we can add a comment with this description?